integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+
+            Document theSequel = new Document();
+            theSequel.setId(42L);
+            theSequel.setName("test2");
+            theSequel.setOwner(person);
+            entityManager.merge(theSequel);
+        });
+
+        doInJPA(entityManager -> {
+            List<Document> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .union(
+                        new BlazeJPAQuery<Document>().intersect(
+                                select(document).from(document).where(document.id.eq(41L)),
+                                new BlazeJPAQuery<Document>().except(
+                                        select(document).from(document).where(document.id.eq(42L)),
+                                        select(document).from(document).where(document.id.eq(43L)))),
+                        select(document).from(document).where(document.id.eq(46L)))
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoMySQL.class, NoFirebird.class, NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class })
+    public void testComplexSubqueryUnion() {
+        doInJPA(entityManager -> {
+            Person person = new Person();
+            person.setName("Person");
+            entityManager.persist(person);
+
+            Document theBook = new Document();
+            theBook.setId(1337L);
+            theBook.setName("test");
+            theBook.setOwner(person);
+            entityManager.merge(theBook);
+
+            Document theSequel = new Document();
+            theSequel.setId(42L);
+            theSequel.setName("test2");
+            theSequel.setOwner(person);
+            entityManager.merge(theSequel);
+        });
+
+        doInJPA(entityManager -> {
+            SetExpression<Long> union = new BlazeJPAQuery<Long>(entityManager, cbf)
+                    .union(select(document.id).from(document).where(document.id.eq(1337L)),
+                            new BlazeJPAQuery<Long>().intersect(
+                                    select(document.id).from(document).where(document.id.eq(41L)),
+                                    new BlazeJPAQuery<Long>().except(
+                                            select(document.id).from(document).where(document.id.eq(42L)),
+                                            select(document.id).from(document).where(document.id.eq(43L)))),
+                            select(document.id).from(document).where(document.id.eq(46L))
+                    );
+
+            QDocument book2 = new QDocument("secondBook");
+
+            List<Document> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .select(book2)
+                    .from(book2).where(book2.id.in(union))
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class })
```

Comment:
Add a comment like e.g. `// NOTE: No advanced sql support for datanucleus, eclipselink and openjpa yet`
Old MySQL version should work fine because we normally inline the CTE. Any reason to disable this here?

Reply:
I'll run it again